AgniMessenger/src/test/AgniTestUtilities.java

```diff
@@ -0,0 +1,13 @@
+package test;
+
+public class AgniTestUtilities {
+    public static byte[] hexStringToByteArray(String s) {
+        int len = s.length();
+        byte[] data = new byte[len / 2];
+        for (int i = 0; i < len; i += 2) {
+            data[i / 2] = (byte) ((Character.digit(s.charAt(i), 16) << 4)
+                                 + Character.digit(s.charAt(i+1), 16));
+        }
```

Comment:
This totally works but rather than doing all this work, you can simple do:
    byte[] data = s.getBytes ( "US-ASCII");
    return data;

The cool thing about that is the string will be encoded in whatever charset you enter as the arg.